src/index.ts

```diff
@@ -312,19 +301,8 @@ const typescript: PluginImpl<RPT2Options> = (options) =>
 		{
 			generateRound = 0; // in watch mode, buildEnd resets generate count just before generateBundle for each output
 
-			if (err)
-			{
-				buildDone();
-				// workaround: err.stack contains err.message and Rollup prints both, causing duplication, so split out the stack itself if it exists (c.f. https://github.com/ezolenko/rollup-plugin-typescript2/issues/103#issuecomment-1172820658)
-				const stackOnly = err.stack?.split(err.message)[1];
-				if (stackOnly)
-					this.error({ ...err, message: err.message, stack: stackOnly });
-				else
-					this.error(err);
-			}
-
-			if (!pluginOptions.check)
-				return buildDone();
+			if (err || !pluginOptions.check)
+			  return buildDone();
```

Comment:
the logic here is a bit confusing -- I had to re-read my own code a few times.
Basically, I made sure the diff was _functionally_ equivalent to the previous one, minus the error rewriting.

`this.error` causes an exit and `return` will similarly end the function.

If there's an error, we just want to call `cache.done()` (as `buildDone()` does) and otherwise let the error proceed.
If we're not doing type-checking, we can return early as the code below is only relevant when type-checking